src/neptune/api.d

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+
+    helper methods to access the octod/github API
+
+    Copyright:
+        Copyright (c) 2009-2016 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.api;
+
+import octod.core;
+import octod.api.repos;
+
+
+/*******************************************************************************
+
+    Returns:
+        global api object to interact with github
+
+*******************************************************************************/
+
+public ref HTTPConnection api ( )
+{
+
+    static HTTPConnection _api;
+    static bool connected = false;
+
+    if (!connected)
+    {
+        _api = HTTPConnection.connect(getConf());
+        connected = true;
+    }
+
+    return _api;
+}
+
+
+/*******************************************************************************
+
+    Returns:
+        neptune configuration. If it doesn't exists, will prompt the user for
+        the required infomation
+
+*******************************************************************************/
+
+public Configuration getConf ( )
+{
+    import neptune.cmd;
+
+    Configuration cfg;
+    cfg.dryRun = false;
+
+    try cfg.oauthToken = cmd("git config neptune.oauth");
```

Comment:
Also, may I suggest `oauthtoken` to match the config in `git hub` (and/or maybe falling back to use `hub.oauthtoken` (maybe as part of the setup if `neptune.oauthtoken` is not found but `hub.oauthtoken` is there, ask if the user want to reuse that token and if they will, store it too as `neptune.oauthtoken`).